ctparse/time/rules.py

```diff
@@ -779,3 +782,184 @@ def _adjust_h(t: Time) -> Optional[int]:
+    r"(?P<d_{}>{}\b)".format(dur.value, expr) for dur, expr in _durations
+)
+_rule_durations = r"({})\s*".format(_rule_durations)
+
+# Rules regarding durations
+@rule(r"(?P<num>\d+)\s*" + _rule_durations)
+def ruleDigitDuration(ts: datetime, m: RegexMatch) -> Optional[Duration]:
+    # 1 day, 1 night etc.
+    num = m.match.group("num")
+    if num:
+        for n, _, in _durations:
+            unit = m.match.group("d_" + n.value)
+            if unit:
+                return Duration(int(num), n)
+
+    return None
+
+
+@rule(_rule_named_number + _rule_durations)
+def ruleNamedNumberDuration(ts: datetime, m: RegexMatch) -> Optional[Duration]:
+    # one day, two nights, thirty days etc.
+    num = None
+    for n, _ in _named_number:
+        match = m.match.group("n_{}".format(n))
+        if match:
+            num = n
+            continue
+
+    if num:
+        for d, _, in _durations:
+            unit = m.match.group("d_" + d.value)
+            if unit:
+                return Duration(num, d)
+
+    return None
+
+
+@rule(r"(hal[fb]e?|1/2)(\s+an?)?\s*" + _rule_durations)
+def ruleDurationHalf(ts: datetime, m: RegexMatch) -> Optional[Duration]:
+    # half day, half hour, 1/2 hour
+    for n, _, in _durations:
+        if m.match.group("d_" + n.value):
+            if n == DurationUnit.HOURS:
+                return Duration(30, DurationUnit.MINUTES)
+            if n == DurationUnit.DAYS:
+                return Duration(12, DurationUnit.HOURS)
+
+    return None
+
+
+@rule(dimension(Interval), r"f[üo]r", dimension(Duration))
+def ruleIntervalConjDuration(
+    ts: datetime, interval: Interval, _: RegexMatch, dur: Duration
+) -> Optional[Interval]:
+    # Example: people tend to repeat themselves when specifying durations
+    # 15-16 Nov für 1 Nacht
+    return ruleDurationInterval(ts, dur, interval)  # type: ignore
+
+
+@rule(dimension(Interval), dimension(Duration))
+def ruleIntervalDuration(
+    ts: datetime, interval: Interval, dur: Duration
+) -> Optional[Interval]:
+    # Variant without conjunction
+    # 15-16 Nov 1 Nacht
+    return ruleDurationInterval(ts, dur, interval)  # type: ignore
+
+
+@rule(dimension(Duration), dimension(Interval))
+def ruleDurationInterval(
+    ts: datetime, dur: Duration, interval: Interval
+) -> Optional[Interval]:
+    # 3 days 15-18 Nov
+    if interval.isDateInterval:
+        delta = interval.t_to.dt - interval.t_from.dt
+        dur_delta = _duration_to_relativedelta(dur)
+        if delta.days == dur_delta.days:
+            return interval
+    return None
+
+
+@rule(dimension(Time), r"f[üo]r", dimension(Duration))
+def ruleTimeDuration(
+    ts: datetime, t: Time, _: RegexMatch, dur: Duration
+) -> Optional[Interval]:
+    # Examples:
+    # on the 27th for one day
+    # heute eine Übernachtung
+    if t.hasDate:
+        # To make an interval we should at least have a date
+        if (
+            dur.unit
+            in (
+                DurationUnit.DAYS,
```

Comment:
Maybe we should/could add some predicates on the `Duration` as well. That's what I thought on the `ruleDurationInterval` above. I haven't thought long about this, but they seem to mostly make sense for `predicate("isDateInterval") predicate("isDayNightDuration")` sequences OR `predicate("isTime") predicate("isHourOrMinuteDuration")` (clearly these are fake predicates that would need to be defined). Using predicates is a bit more expensive, but also reduces the number of branches. Not sure though whether this is super important with Durations as they are less common.